test/integration/change-streams/change_stream.test.ts

```diff
@@ -1767,7 +1718,44 @@ describe('ChangeStream resumability', function () {
             expect(aggregateEvents).to.have.lengthOf(2);
           }
         );
+
+        it(
+          `supports consecutive resumes on error code ${code} ${error}`,
```

Comment:
Just a note that these test additions will likely result in merge conflicts for https://github.com/mongodb/node-mongodb-native/pull/3454 and we should be extra careful in that PR, once it's rebased, that all these tests are pulled in